analyses/cell-type-ewings/template_notebooks/auc-singler-workflow/03-clustering.Rmd

```diff
@@ -0,0 +1,265 @@
+
+```{r}
+# cluster purity for different params
+plot_cluster_stats(all_stats_df, purity)
+```
+
+### Cluster stability 
+
+```{r, warning=FALSE, message=FALSE}
+# calculate cluster stability
+stability_df <- get_cluster_stability(sce, all_cluster_results) |>
+  dplyr::mutate(
+    k_value = as.factor(k_value),
+    # make sure that k = 5 comes first
+    k_value = forcats::fct_relevel(k_value, "5", after = 0)
+  )
+
+# plot stability across all values of k
+ggplot(stability_df, aes(x = k_value, y = ari)) +
+  geom_jitter(width = 0.1) +
+  labs(title = "Cluster stability") +
+  stat_summary(
+    aes(group = k_value),
+    color = "red",
+    # median and quartiles for point range
+    fun = "median",
+    fun.min = function(x) {
+      quantile(x, 0.25)
+    },
+    fun.max = function(x) {
+      quantile(x, 0.75)
+    }
+  )
+```
+
+
+## Compare clusters to cell types from `SingleR`
+
+Now we will compare the clustering results to the cell type assignments obtained from `SingleR` in the `aucell-singler-annotation.sh` workflow. 
+To compare results we will calculate the Jaccard similarity index between clusters and cell types. 
+We expect that good clustering will line up with the cell type assignments so that there is ~ 1 cell type per cluster. 
+
+For the plots, we will only display the top 7 cell types identified by `SingleR` and all other cells will be lumped together into `All remaining cell types`. 
+
+```{r}
+cluster_classification_df <- all_cluster_results |>
+  # add in classifications from singler
+  dplyr::left_join(singler_classification_df, by = "barcodes") |>
+  dplyr::mutate(
+    # first grab anything that is tumor and label it tumor
+    # NA should be unknown
+    singler_annotation = dplyr::case_when(
+      stringr::str_detect(singler_annotation, "tumor") ~ "tumor",
+      is.na(singler_annotation) ~ "unknown", # make sure to separate out unknown labels
+      .default = singler_annotation
+    ) |>
+      forcats::fct_relevel("tumor", after = 0),
+    # get the top cell types for plotting later
+    singler_lumped = singler_annotation |>
+      forcats::fct_lump_n(7, other_level = "All remaining cell types", ties.method = "first") |>
+      forcats::fct_infreq() |>
+      forcats::fct_relevel("All remaining cell types", after = Inf),
+    nn_param = forcats::fct_relevel(nn_param, "k_5", after = 0)
+  )
+```
+
+```{r, fig.height=10, fig.width=7}
+# get heatmap for showing cluster assignments vs. singler assignments
+cluster_celltype_heatmap(cluster_classification_df)
+```
+
+
+## Marker gene expression across clusters 
+
+Finally, we will look at the expression of marker genes across each cluster for each value of K. 
+In these plots, each row shows the distribution of the specified marker genes in that cluster. 
+Each panel is labeled with the marker gene expression being plotted. 
+
+```{r}
+# read in marker genes table
+marker_genes_df <- readr::read_tsv(params$marker_genes_file, show_col_types = FALSE) |>
```

Comment:
I don't think you need this since you set the option at the top

```suggestion
marker_genes_df <- readr::read_tsv(params$marker_genes_file) |>
```